src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/services/service-form/service-form.component.spec.ts

```diff
@@ -251,6 +252,17 @@ describe('ServiceFormComponent', () => {
         const ssl_key = fixture.debugElement.query(By.css('#ssl_key'));
         expect(ssl_key).toBeNull();
       });
+
+      it('should test .pem file', () => {
+        const pemCert = expandN(
+          /^-----BEGIN .+?-----(\n|\r|\f)((.+)?((\n|\r|\f).+)*)(\n|\r|\f)-----END .+?-----[\n\r\f]*$/,
+          1
+        );
```

Comment:
If you use the same regex to generate patterns that match that expression... aren't you cheating yourself?

The proper unit test would be using a/few real RPM file/s and match this against those...